### PR TITLE
Fix create torrent out-of-range piece size

### PR DESCRIPTION
**Pull request by Coeur**
Fix a UI glitch when you attempt to set piece size too high or too low.

### CODE REVIEW

File: macosx/CreatorWindowController.mm [hunk 1]
```diff
@@ -360,7 +360,20 @@ - (IBAction)cancelCreateProgress:(id)sender
 
 - (IBAction)incrementOrDecrementPieceSize:(id)sender
 {
-    auto const piece_size = static_cast<uint32_t>(pow(2.0, [sender intValue]));
+    auto power = [(NSStepper*)sender intValue];
```
Comment (nevack) [inline macosx/CreatorWindowController.mm:363]: Can't we just set min/max value on NSStepper itself?
https://developer.apple.com/documentation/appkit/nsstepper

File: macosx/CreatorWindowController.mm [hunk 2]
```diff
@@ -360,7 +360,20 @@ - (IBAction)cancelCreateProgress:(id)sender
...
+        [(NSStepper*)sender setIntValue:31];
+        return;
+    }
+    if (power <= 13)
+    {
+        // below min piece size
+        [(NSStepper*)sender setIntValue:14];
+        return;
+    }
+    auto const piece_size = uint32_t(1) << power;
```
Comment (nevack) [inline macosx/CreatorWindowController.mm:376]: nit: `uint32_t(1)` -> `1U`?

Comment (Coeur) [reply macosx/CreatorWindowController.mm]: OK. Can you give a second look now, @nevack ? :)

**Review by nevack — APPROVED** [review]
LGTM

**Comment by Coeur** [comment]
Notes: Fix a UI glitch in macOS app when you attempt to set piece size too high or too low.

**Pull request merged**